src/dmd/backend/barray.d

```diff
@@ -110,6 +110,4 @@ unittest
         v = cast(int) i * 2;
     foreach (i, ref const v; a[])
         assert(v == i * 2);
-    a.dtor();
-    assert(a.length == 0);
```

Comment:
Why was this assertion removed?

Reply:
because you can't assert on effects of the destructors, only side effects such as writing to globals.